src/NLog/LayoutRenderers/GuidLayoutRenderer.cs

```diff
@@ -59,13 +59,34 @@ public GuidLayoutRenderer()
         public string Format { get; set; }
 
         /// <summary>
+        /// Generate the Guid from the NLog LogEvent (Will be the same for all targets)
+        /// </summary>
+        [DefaultValue(false)]
+        public bool LogEventGuid { get; set; }
+
+        /// <summary>
         /// Renders a newly generated GUID string and appends it to the specified <see cref="StringBuilder" />.
         /// </summary>
         /// <param name="builder">The <see cref="StringBuilder"/> to append the rendered data to.</param>
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
-            builder.Append(Guid.NewGuid().ToString(this.Format));
+            if (this.LogEventGuid)
+            {
+                byte d = (byte)(LogEventInfo.ZeroDate.Ticks >> 56);
+                byte e = (byte)(LogEventInfo.ZeroDate.Ticks >> 48);
+                byte f = (byte)(LogEventInfo.ZeroDate.Ticks >> 40);
+                byte g = (byte)(LogEventInfo.ZeroDate.Ticks >> 32);
+                byte h = (byte)(LogEventInfo.ZeroDate.Ticks >> 24);
+                byte i = (byte)(LogEventInfo.ZeroDate.Ticks >> 16);
+                byte j = (byte)(LogEventInfo.ZeroDate.Ticks >> 8);
+                byte k = (byte)(LogEventInfo.ZeroDate.Ticks);
+                builder.Append(new Guid(logEvent.SequenceID, (short)logEvent.TimeStamp.Month, (short)logEvent.TimeStamp.Year, d, e, f, g, h, i, j, k).ToString(this.Format));
```

Comment:
It is global unique identifier of a LogEvent within the running application. But probably not global in the universe.